src/Materials/Textures/Loaders/basisTextureLoader.ts

```diff
@@ -0,0 +1,100 @@
+     */
+    public transformUrl(rootUrl: string, textureFormatInUse: Nullable<string>): string {
+        return rootUrl;
+    }
+
+    /**
+     * Gets the fallback url in case the load fail. This can return null to allow the default fallback mecanism to work
+     * @param rootUrl the url of the texture
+     * @param textureFormatInUse defines the current compressed format in use iun the engine
+     * @returns the fallback texture
+     */
+    public getFallbackTextureUrl(rootUrl: string, textureFormatInUse: Nullable<string>): Nullable<string> {
+        return null;
+    }
+
+    /**
+     * Uploads the cube texture data to the WebGl Texture. It has alreday been bound.
+     * @param data contains the texture data
+     * @param texture defines the BabylonJS internal texture
+     * @param createPolynomials will be true if polynomials have been requested
+     * @param onLoad defines the callback to trigger once the texture is ready
+     * @param onError defines the callback to trigger in case of error
+     */
+    public loadCubeData(data: string | ArrayBuffer | (string | ArrayBuffer)[], texture: InternalTexture, createPolynomials: boolean, onLoad: Nullable<(data?: any) => void>, onError: Nullable<(message?: string, exception?: any) => void>): void {
+        throw ".basis not supported in Cube.";
+    }
+
+    /**
+     * Uploads the 2D texture data to the WebGl Texture. It has alreday been bound once in the callback.
+     * @param data contains the texture data
+     * @param texture defines the BabylonJS internal texture
+     * @param callback defines the method to call once ready to upload
+     */
+    public loadData(data: ArrayBuffer, texture: InternalTexture,
+        callback: (width: number, height: number, loadMipmap: boolean, isCompressed: boolean, done: () => void) => void): void {
+        // Verify Basis Module is loaded and detect file info and format
+        BasisTools.VerifyBasisModule();
+        var loadedFile = BasisTools.LoadBasisFile(data);
+        var fileInfo = BasisTools.GetFileInfo(loadedFile);
+        var format = BasisTools.GetSupportedTranscodeFormat(texture.getEngine(), fileInfo);
+
+        // TODO this should be done in web worker
+        var transcodeResult = BasisTools.TranscodeFile(format, fileInfo, loadedFile);
+
+        // Upload data to texture
+        callback(fileInfo.width, fileInfo.height, false, true, () => {
+            if (transcodeResult.fallbackToRgb565) {
+                // Load rgb565Data to texture if conversion is needed
+                texture.getEngine()._unpackFlipY(texture.invertY);
+                var gl = texture.getEngine()._gl;
+                gl.texImage2D(gl.TEXTURE_2D, 0, gl.RGB, fileInfo.alignedWidth, fileInfo.alignedHeight, 0, gl.RGB, gl.UNSIGNED_SHORT_5_6_5, transcodeResult.pixels);
+
+                // TODO this is not working because computed power of 2 image size is larger than rgb565Data causing an error, working around this with below code

```

Comment:
Any ideas on how to support this?

Reply:
Not sure to understand why it is different from a regular texture at that point? We do the same rescale for regular jpg texture for instance when we need pow 2

Reply:
like here: https://github.com/BabylonJS/Babylon.js/blob/master/src/Engines/engine.ts#L4312